engine/src/org/pentaho/di/www/TransformationMap.java

```diff
@@ -41,18 +46,15 @@
  *
  */
 public class TransformationMap {
-  private Map<CarteObjectEntry, Trans> transformationMap;
-  private Map<CarteObjectEntry, TransConfiguration> configurationMap;
+  private final Map<CarteObjectEntry, TransData> transMap;
 
-  private Map<String, List<SocketPortAllocation>> hostServerSocketPortsMap;
+  private final Map<String, List<SocketPortAllocation>> hostServerSocketPortsMap;
```

Comment:
Hello ,@mbatchelor . Here we can't change to Deque implementation as https://github.com/ArtsiomYurhevich/pentaho-kettle/blob/d12f1978ce5c1625c5e862975695c64d98b2f922/engine/src/org/pentaho/di/www/TransformationMap.java#L255 uses set by index. And i have left synchronization blocks due to consistent synchronization over operations with boolean value isAllocated. Look https://github.com/ArtsiomYurhevich/pentaho-kettle/blob/d12f1978ce5c1625c5e862975695c64d98b2f922/engine/src/org/pentaho/di/www/TransformationMap.java#L234#L240 , https://github.com/ArtsiomYurhevich/pentaho-kettle/blob/d12f1978ce5c1625c5e862975695c64d98b2f922/engine/src/org/pentaho/di/www/TransformationMap.java#L265#L269 and deallocate methods which set the value to false. Without synchronization it can be possible such cases as allocating already allocated SocketPortAllocation and etc. Also List implementation CopyOnWriteArrayList is  a thread safe one https://github.com/ArtsiomYurhevich/pentaho-kettle/blob/d12f1978ce5c1625c5e862975695c64d98b2f922/engine/src/org/pentaho/di/www/TransformationMap.java#L210 and we can use it in this case.